src/script/lua_api/l_mapgen.cpp

```diff
@@ -1557,6 +1560,10 @@ int ModApiMapgen::l_generate_ores(lua_State *L)
 	mg.vm   = checkObject<LuaVoxelManip>(L, 1)->vm;
 	mg.ndef = emerge->ndef;
 
+	auto *bgen = getBiomeGen(L);
+	if (bgen && bgen->getType() == BIOMEGEN_ORIGINAL)
```

Comment:
does the type matter here? I don't see any casts to `BiomeGenOriginal`

Reply:
I'm not sure to completely understand what `BiomeGenOriginal` is, so I replicated codes from other places. Is that some "biome system type" that is useless for now but is here because other biome systems *could* exist? or does it have another use?